creusot-contracts/src/local_invariant.rs

```diff
@@ -0,0 +1,172 @@
+//! Define local invariants
+//!
+//! [Local invariants](LocalInvariant) are not the same as [invariants](Invariant): they
+//! allow the use of a shared piece of data to be used in the invariant (see
+//! [`LocalInvariantSpec`]), but in return they impose a much more restricted access to
+//! the underlying data, as well as the use of [`Namespaces`].
```

Comment:
Can you expand on their use a bit more in the docs? this seems _very_ cool but I can't quite grasp what it would be good for / where it would be limited.

Reply:
Yeah, I added an example in the docs and the test suite. I also kind of want to showcase how to use this with ghost resources, but I fear this will get complex quickly (and there is already the 'Persistent arrays' case study in this case).